sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -155,9 +155,15 @@ def rank_error(msg):
               or str(response.currentState) == 'JOB_STATE_UPDATED'
               or str(response.currentState) == 'JOB_STATE_DRAINED'):
             break
-          # The job has failed; ensure we see any final error messages.
-          sleep_secs = 1.0      # poll faster during the final countdown
-          final_countdown_timer_secs -= sleep_secs
+
+          # Check that job is in a post-preparation state before starting the
+          # final countdown.
+          if (str(response.currentState) not in (
+              'JOB_STATE_PENDING', 'JOB_STATE_QUEUED')):
```

Comment:
I encountered https://stackoverflow.com/questions/68110323/encounter-assertionerror-job-did-not-reach-to-a-terminal-state-after-waiting-in/68120786#68120786 that eventually led me back here. Is it possible that a new pre-running state has been added since this change? (the question does not specify version, but I assume from context clues that it is new enough to have this change in it)

Reply:
Another possibly related instance: https://stackoverflow.com/questions/68128465/is-it-possible-to-stop-the-job-in-starting-state

I'll file a Jira and follow up, I just want to note for posterity.

Reply:
It does not look like another pre-running state was added. (See for reference: https://cloud.google.com/dataflow/docs/reference/rest/v1b3/projects.jobs)

> Another possibly related instance: stackoverflow.com/questions/68128465/is-it-possible-to-stop-the-job-in-starting-state

According to the same link:
pending -> cancelling is not a valid transition
queued -> cancelling is a valid transition

(But the doc might also be out of date.)

Thank you for filing the jira. Please share jira id here for reference.